tests/middleware.test.js

```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import errorHandler from '../src/middlewares/errorHandler';
+
+describe('Error Handler', () => {
+  let req, res, next;
+  beforeEach(() => {
+    res = {
+      status(code) {
+        res.status = code;
+      },
+      json() {}
+    };
+    req = {};
+    next = () => {};
+  });
+  it('should return 500 error status', async () => {
+    errorHandler(new Error('Test Error'), req, res, next);
+    expect(res.status).to.be.eql(500);
+  });
+  it('should return specified error for production', async () => {
+    process.env.NODE_ENV = 'production';
+    const err = new Error('403 Error');
+    err.status = 403;
+    errorHandler(err, req, res, next);
+    expect(res.status).to.be.eql(403);
+  });
+  it('should return specified error for development', async () => {
```

Comment:
Is there a way to test that it returns the stack trace for development?

Reply:
I'm still levelling up on sinon, so for now, I only tested the error handler just as a function not as a request.

Reply:
Yeah, Sinon is actually for functions, not requests. I mean testing that `res.json.errorMessage` is not an empty object in development and is an empty object in production i.e.

In development,
```javascript
expect(res.json.errorMessage).not.to.be.eql({});
```

In production,
```javascript
expect(res.json.errorMessage).to.be.eql({});
```